### PR TITLE
Sidebar (filter) improved and scripts refactor.

### PR DESCRIPTION
**Pull request by Maksims**
General refactor of scripts and sidebar filters.

### Functionality improvements:

1. Sidebar new is visually separated using slight shadow, improves on tablet sizes and when scrollbar in sidebar is not visible.
2. Filters input field now can be cleared using X button.
3. Filters input once changed will stay in URL, so it can be shared with such filter. (tried history push state, but felt weird)
4. On page load sidebar will scroll to current link.
5. Code elements are now scrollable.

### CODE REVIEW

File: tmpl/page.tmpl [hunk 1]
```diff
@@ -2,12 +2,14 @@
 <html lang="en">
     <head>
         <title>{{title}} | PlayCanvas API Reference</title>
-        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
```
Comment (willeastcott) [inline tmpl/page.tmpl:5]: Lighthouse complains about this:
https://web.dev/meta-viewport/
Mobile Safari actually ignores this setting. But yeah, I would recommend deleting it as it seems to be considered 'bad practice'.
If you haven't used Lighthouse before (in Chrome Dev Tools), check it out. It's awesome and I've used it to tune all the marketing pages in playcanvas.com

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
Thanks - this is awesome 🚀

**Comment by yaustar** [comment]
The sidebar for the API docs no longer filter when typing. Was this PR related?
Kapture 2020-12-25 at 12 18 15

**Comment by Maksims** [comment]
Only on landing page.
Should be fixed now: https://github.com/playcanvas/playcanvas-jsdoc-template/pull/25#event-4148873309

**Comment by willeastcott** [comment]
@yaustar The deployment job for the prod API ref was building from master and not stable. Whoops. I've updated the job to always use stable and redeployed.

**Pull request merged**